### PR TITLE
Mainboard firmware architecture description

### PR DESCRIPTION
**Pull request by npetersen2**
This PR adds a markdown doc which describes the AMDS mainboard firmware design. This will be helpful to future users who want to understand the system performance.

[Read the main doc I added here.](https://github.com/Severson-Group/AMDS/blob/npetersen2-patch-1/Mainboard/Firmware/FirmwareArchitecture.md)

### CODE REVIEW

File: Mainboard/Firmware/FirmwareArchitecture.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+### Interface to Master
+
+The AMDS firmware is designed to interface to the master controller over four logical wires: 2x TX and 2x RX. Physically, these signals are all diff pairs for noise immunity.
+
+<img src="./images/firmware_arch_interface.svg" width="500" />
+
+#### RX Signal: `SYNC_ADC`
+
+One of the two signals, `SYNC_ADC`, is used to trigger ADC sampling. Normally, the signal is a square wave. On each edge, the AMDS samples all the sensor cards on the mainboard. Normally, the master triggers a transition on this RX signal when the PWM carrier is at a peak or valley. This synchronizes the ADC sampling to the inverter PWM, thus reducing sampling noise.
```
Comment (Bharat-Ramadas) [inline Mainboard/Firmware/FirmwareArchitecture.md]: Are both rising and falling edges recognized as interrupts? Might be a good idea to clarify that here.

**Review by ngadiyar93 — COMMENTED** [review]
Thanks @npetersen2. This is very useful. 
After reading this, it is a lot clear on how the AMDS interfaces with the AMDC (or any other platform).

I think it may be useful if we could add details somewhere on how to configure the AMDC to read data from the AMDS for a typical 3 phase inverter (the AMDS has 3 current cards and a DC bus voltage sensor). 
That could be part of this doc or a separate document in AMDS / AMDC Examples / Group KB. Thoughts?

**Comment by GnomeTek** [comment]
Suggestion of updating Sync signals to " 'Sync_xxx' shall be a square wave". A square wave/pulse is a requirement for proper timing, right? 
The use of "normally" suggest I could feed this a triangle wave and the system would maintain the described timing latency. 

For TX signals 'DATAx' the UART is described, how is the ADC data packed into the 8bit transmission? is ADC data 8bit, 32bit? Byte order in UART transmission might be helpful also.

File: Mainboard/Firmware/FirmwareArchitecture.md (hunk 1 above)
Comment (npetersen2) [reply Mainboard/Firmware/FirmwareArchitecture.md]: See updated doc

**Comment by npetersen2** [comment]
@ngadiyar93 Good idea for adding a doc about the AMDC side of the AMDS interface.

I think this doesn't belong in this repo, but instead, in the AMDC repos -- probably `AMDC-Firmware` since there is a full driver in the FPGA to interface to the AMDS.

Can you create an issue in `AMDC-Firmware` which explains the need for this? Thanks.

**Comment by npetersen2** [comment]
@GnomeTek See updated doc. I added comments on data packing/ordering in UART stream.

**Comment by npetersen2** [comment]
This ended up getting merged into the docs.amd.dev website, see: https://docs.amdc.dev/accessories/amds/firmware/

**Pull request closed without merging**